src/test/java/io/cdap/plugin/splunk/source/batch/SplunkBatchSourceConfigTest.java

```diff
@@ -0,0 +1,489 @@
+      }
+    };
+
+    MockFailureCollector collector = new MockFailureCollector(MOCK_STAGE);
+    config.validate(collector);
+    Assert.assertTrue(collector.getValidationFailures().isEmpty());
+  }
+
+  @Test
+  public void testInvalidUrl() {
+    SplunkBatchSourceConfig config = new SplunkBatchSourceConfig(
+      "reference",
+      "basic",
+      "apiToken",
+      "userName",
+      60000,
+      60000,
+      3,
+      60000,
+      100,
+      "invalid",
+      "password",
+      "executionMode",
+      "outputFormat",
+      "searchString",
+      "searchId",
+      0L,
+      "earliestTime",
+      "latestTime",
+      "indexedEarliestTime",
+      "indexedLatestTime",
+      100L,
+      "schema");
+
+    MockFailureCollector collector = new MockFailureCollector(MOCK_STAGE);
+    config.validate(collector);
+
+    Assert.assertEquals(1, collector.getValidationFailures().size());
+    List<ValidationFailure.Cause> causeList = collector.getValidationFailures().get(0).getCauses();
+    Assert.assertEquals(1, causeList.size());
+    Assert.assertEquals(SplunkBatchSourceConfig.PROPERTY_URL, collector.getValidationFailures().get(0)
+      .getCauses().get(0).getAttribute(CauseAttributes.STAGE_CONFIG));
+  }
+
+  @Test
+  public void testUsernameEmpty() {
+    SplunkBatchSourceConfig config = new SplunkBatchSourceConfig(
+      "reference",
+      "basic",
+      "apiToken",
+      "",
+      60000,
+      60000,
+      3,
+      60000,
+      100,
+      "https://localhost:8089",
+      "password",
+      "executionMode",
+      "outputFormat",
+      "searchString",
+      "searchId",
+      0L,
+      "earliestTime",
+      "latestTime",
+      "indexedEarliestTime",
+      "indexedLatestTime",
+      100L,
+      "schema");
+
+    MockFailureCollector collector = new MockFailureCollector(MOCK_STAGE);
+    config.validate(collector);
+
+    Assert.assertEquals(1, collector.getValidationFailures().size());
+    List<ValidationFailure.Cause> causeList = collector.getValidationFailures().get(0).getCauses();
+    Assert.assertEquals(1, causeList.size());
+    Assert.assertEquals(BaseSplunkConfig.PROPERTY_USERNAME, collector.getValidationFailures().get(0)
+      .getCauses().get(0).getAttribute(CauseAttributes.STAGE_CONFIG));
+  }
+
+  @Test
+  public void testPasswordEmpty() {
+    SplunkBatchSourceConfig config = new SplunkBatchSourceConfig(
+      "reference",
+      "basic",
+      "apiToken",
+      "userName",
+      60000,
+      60000,
+      3,
+      60000,
+      100,
+      "https://localhost:8089",
+      "",
+      "executionMode",
+      "outputFormat",
+      "searchString",
+      "searchId",
+      0L,
+      "earliestTime",
+      "latestTime",
+      "indexedEarliestTime",
+      "indexedLatestTime",
+      100L,
+      "schema");
+
+    MockFailureCollector collector = new MockFailureCollector(MOCK_STAGE);
+    config.validate(collector);
+
+    Assert.assertEquals(1, collector.getValidationFailures().size());
```

Comment:
try to use utility class as we do in other plugins to avoid code duplication in tests

Reply:
Done.